src/gurt/dlog.c

```diff
@@ -296,6 +310,55 @@ static void dlog_cleanout(void)
 #endif
 }
 
+void *d_log_tls_get()
+{
+	if (!mst.tls_key_created)
+		return NULL;
+
+	return pthread_getspecific(mst.tls_key);
+}
+
+int d_log_tls_init()
+{
+	void		*tls;
+	int		 rc;
+
+	if (!mst.tls_key_created)
+		return -1;
+
+	tls = d_log_tls_get();
+	if (tls != NULL)
+		return 0;
+
+	rc = posix_memalign(&tls, 16, sizeof(struct clog_tls));
+	if (rc || tls == NULL)
+		return -1;
+
+	rc = pthread_setspecific(mst.tls_key, tls);
+	if (rc) {
+		free(tls);
+		return -1;
+	}
+
+	return 0;
+}
+
+void d_log_tls_fini()
```

Comment:
(style)  Bad function definition - void d_log_tls_fini() should probably be void d_log_tls_fini(void)